### PR TITLE
make "can this be a boot partition" use v2 capabilities

### PR DESCRIPTION
**Pull request by mwhudson**
Turns out I had half of this sitting around in a branch I'd forgotten about.

### CODE REVIEW

File: subiquity/common/filesystem/tests/test_manipulator.py [hunk 1]
```diff
@@ -393,32 +420,93 @@ def DONT_test_add_boot_BIOS_preserved(self):  # needs v2 partitioning
...
+        elif bl == Bootloader.PREP:
+            return sizes.PREP_GRUB_SIZE_BYTES
+        else:
+            self.fail("unexpected bootloader %r" % (bl,))
+
+    @parameterized.expand([
+        (Bootloader.UEFI, 1),
+        (Bootloader.PREP, 2),
+        (Bootloader.UEFI, 1),
+        (Bootloader.PREP, 2)])
```
Comment (dbungert) [inline subiquity/common/filesystem/tests/test_manipulator.py:486]: Was this intended as below?
See also 6 repeated copies of this table later in the code
```suggestion
        (Bootloader.UEFI, 1),
        (Bootloader.PREP, 1),
        (Bootloader.UEFI, 2),
        (Bootloader.PREP, 2)])
```

**Review by dbungert — CHANGES_REQUESTED** [review]
Easy test thing to resolve then it should be ready.

File: subiquity/common/filesystem/boot.py [hunk 2]
```diff
@@ -156,42 +156,51 @@ def get_boot_device_plan_bios(device) -> Optional[MakeBootDevicePlan]:
...
+                    ])
+            else:
+                return None
+        elif pg.preserve:
+            break
+        else:
+            movable.append(pg)
+
+    if not movable:
+        return None
```
[the remark this answers is not in the document]
Comment (mwhudson) [reply subiquity/common/filesystem/boot.py:189]: Yeah, I think all of those should only call that if they are sure that the call is OK but it's a bit implicit to be sure.

File: subiquity/common/filesystem/tests/test_manipulator.py (hunk 1 above)
Comment (mwhudson) [reply subiquity/common/filesystem/tests/test_manipulator.py:486]: Ah haha oops.

File: subiquity/common/filesystem/tests/test_manipulator.py [hunk 3]
```diff
@@ -306,8 +310,9 @@ def assertPartitionOperations(self, disk, *ops):
             if new_parts:
                 self.fail("no assertion about {}".format(new_parts))
 
-    def test_add_boot_BIOS_empty(self):
-        manipulator = make_manipulator(Bootloader.BIOS)
+    @parameterized.expand([(1,), (2,)])
```
[the remark this answers is not in the document]
Comment (mwhudson) [reply subiquity/common/filesystem/tests/test_manipulator.py:313]: I'm not super thrilled with it in some ways, like the generated test cases don't get very explicable names and I'd prefer to be able to indicate we should test the cross product of the possibilities (and thus avoid my embarrassing mistake below) but it's sure better than nothing.

pytest has native support for parameterization but it doesn't work in unittest.TestCase subclasses and while I think all our CI bits now run the tests with pytest, we don't actually directly depend on pytest for anything -- maybe we should have a conversation about that.

**Review by dbungert — APPROVED** [review]

**Pull request merged**